io-engine/tests/snapshot_nexus.rs

```diff
@@ -1282,3 +1249,135 @@ async fn test_replica_listing_with_query() {
+    let ms_repl_3 = conn.grpc_handle_shared("ms_repl_3").await.unwrap();
+    // Create Pool-1 and Replica-1.
+    let mut pool_1 = PoolBuilder::new(ms_repl_1.clone())
+        .with_name("pool1")
+        .with_new_uuid()
+        .with_malloc("mem1", POOL_SIZE);
+
+    let mut repl_1 = ReplicaBuilder::new(ms_repl_1.clone())
+        .with_pool(&pool_1)
+        .with_name("r1")
+        .with_new_uuid()
+        .with_size_mb(REPL_SIZE)
+        .with_thin(true);
+
+    pool_1.create().await.unwrap();
+    repl_1.create().await.unwrap();
+    repl_1.share().await.unwrap();
+
+    // Create Pool-2 and Replica-2.
+    let mut pool_2 = PoolBuilder::new(ms_repl_2.clone())
+        .with_name("pool2")
+        .with_new_uuid()
+        .with_malloc("mem2", POOL_SIZE);
+
+    let mut repl_2 = ReplicaBuilder::new(ms_repl_2.clone())
+        .with_pool(&pool_2)
+        .with_name("r2")
+        .with_new_uuid()
+        .with_size_mb(REPL_SIZE)
+        .with_thin(true);
+
+    pool_2.create().await.unwrap();
+    repl_2.create().await.unwrap();
+    repl_2.share().await.unwrap();
+    // Create Pool-3 and Replica-3.
+    let mut pool_3 = PoolBuilder::new(ms_repl_3.clone())
+        .with_name("pool3")
+        .with_new_uuid()
+        .with_malloc("mem3", POOL_SIZE);
+
+    let mut repl_3 = ReplicaBuilder::new(ms_repl_3.clone())
+        .with_pool(&pool_3)
+        .with_name("r3")
+        .with_new_uuid()
+        .with_size_mb(REPL_SIZE)
+        .with_thin(true);
+
+    pool_3.create().await.unwrap();
+    repl_3.create().await.unwrap();
+    repl_3.share().await.unwrap();
+
+    // Create nexus.
+    let mut nex_0 = NexusBuilder::new(ms_nexus.clone())
+        .with_name("nexus0")
+        .with_new_uuid()
+        .with_size_mb(REPL_SIZE)
+        .with_replica(&repl_1)
+        .with_replica(&repl_2)
+        .with_replica(&repl_3);
+
+    nex_0.create().await.unwrap();
+    nex_0.publish().await.unwrap();
+
+    let snapshot_params = SnapshotParams::new(
+        Some(String::from("e1")),
+        Some(String::from("p1")),
+        Some(Uuid::new_v4().to_string()),
+        Some(String::from("s1")),
+        Some(Uuid::new_v4().to_string()),
+        Some(Utc::now().to_string()),
+        false,
+    );
+
+    let replicas = vec![
+        NexusCreateSnapshotReplicaDescriptor {
+            replica_uuid: repl_1.uuid(),
+            snapshot_uuid: Some(Uuid::new_v4().to_string()),
+            skip: false,
+        },
+        NexusCreateSnapshotReplicaDescriptor {
+            replica_uuid: repl_2.uuid(),
+            snapshot_uuid: Some(Uuid::new_v4().to_string()),
+            skip: false,
+        },
+        NexusCreateSnapshotReplicaDescriptor {
+            replica_uuid: repl_3.uuid(),
+            snapshot_uuid: Some(Uuid::new_v4().to_string()),
+            skip: false,
+        },
+    ];
+    let snap_list = nex_0
+        .create_nexus_snapshot(snapshot_params, replicas)
```

Comment:
Do we have tests where we specify only 1 replica but the nexus has more? If not we should add, to ensure it won't create without skip true on the other replicas.

Reply:
Added few error case which can be checked directly in dataplane, including this condition.